frontend/components/analysis/AccessButtons.vue

```diff
@@ -0,0 +1,96 @@
+<script setup lang="ts">
+import { type Analysis, type AccessBlobContentReqItem, type AccessBlobContentResp } from '@/openapi/generated/pacta'
+import JSZip from 'jszip'
+
+const { t } = useI18n()
+const { public: { apiServerURL } } = useRuntimeConfig()
+const pactaClient = usePACTA()
+const { getMaybeMe } = useSession()
+const { isAdmin, isSuperAdmin, maybeMeOwnerId } = await getMaybeMe()
+
+interface Props {
+  analysis: Analysis
+}
+const props = defineProps<Props>()
+
+const prefix = 'components/analysis/AccessButtons'
+const statePrefix = `${prefix}[${useStateIDGenerator().id()}]`
+const tt = (key: string) => t(`${prefix}.${key}`)
+
+const canAccessAsPublic = computed(() => props.analysis.artifacts.every((asset) => asset.sharedToPublic))
+const canAccessAsAdmin = computed(() => {
+  if (isAdmin.value || isSuperAdmin.value) {
+    return props.analysis.artifacts.every((asset) => asset.adminDebugEnabled)
+  }
+  return false
+})
+const canAccessAsOwner = computed(() => {
+  if (maybeMeOwnerId.value) {
+    return maybeMeOwnerId.value === props.analysis.ownerId
+  }
+  return false
+})
+const canAccess = computed(() => {
+  return canAccessAsPublic.value || canAccessAsAdmin.value || canAccessAsOwner.value
+})
+const downloadInProgress = useState<boolean>(`${statePrefix}.downloadInProgress`, () => false)
+const doDownload = async () => {
+  downloadInProgress.value = true
+  await pactaClient.accessBlobContent({
+    items: props.analysis.artifacts.map((asset): AccessBlobContentReqItem => ({
+      blobId: asset.blob.id,
+    })),
+  }).then(async (response: AccessBlobContentResp) => {
+    const zip = new JSZip()
+    await Promise.all(response.items.map(
+      async (item): Promise<void> => {
+        const response = await fetch(item.downloadUrl)
+        const data = await response.blob()
```

Comment:
There should be a way to stream the binary response data directly to the zip without loading it into memory first, can you see if that's an easy option?

Reply:
I played around with a few attempts here, without avail. The docs are here, nothing screams out.

https://stuk.github.io/jszip/documentation/api_jszip/file_data.html

Reply:
Took a look, looks like streaming is supported in Node environments, but not in the browser, c'est la vie